src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -845,38 +752,156 @@ private async ValueTask<HttpConnectionBase> GetHttp3ConnectionAsync(HttpRequestM
             }
         }
 
-        public async ValueTask<HttpResponseMessage> SendWithRetryAsync(HttpRequestMessage request, bool async, bool doRequestAuth, CancellationToken cancellationToken)
+        // Returns null if HTTP3 cannot be used.
+        private async ValueTask<HttpResponseMessage?> TrySendUsingHttp3Async(HttpRequestMessage request, bool async, bool doRequestAuth, CancellationToken cancellationToken)
         {
-            int retryCount = 0;
-
-            while (true)
+            // TODO: Replace with Platform-Guard Assertion Annotations once https://github.com/dotnet/runtime/issues/44922 is finished
+            if ((OperatingSystem.IsLinux() && !OperatingSystem.IsAndroid()) || OperatingSystem.IsWindows() || OperatingSystem.IsMacOS())
```

Comment:
The guard (`if ((OperatingSystem ...`) should rather be at the call site and this method should be annotated with:
```
[SupportedOSPlatform("windows")]
[SupportedOSPlatform("linux")]
[SupportedOSPlatform("macos")]
```
As is `GetHttp3ConnectionAsync` for example. Then, the whole method gets trimmed out, which is IMHO what we want here.

Reply:
Thanks, will fix.